main.go

```diff
@@ -374,55 +527,75 @@ func validateBaseURLScheme(baseURL string) (string, error) {
-	scheme := u.Scheme
-	if scheme != "" {
-		return baseURL, nil
-	}
-	URL := "https://" + baseURL
-	return URL, nil
+	u.Scheme = "https"
+	q := u.Query()
+	u.RawQuery = q.Encode()
+	return u.String(), nil
 }
 
 func main() {
 	// Declaring variables for flags
 	var token, baseURL, namespace, project, timeInterval string
 	var advance int
 	// Command Line Parsing Starts
-	flag.StringVar(&token, "Token", "jGWPwqQUuf37b", "Gitlab api key/token")
-	flag.StringVar(&timeInterval, "TimeInterval", "daily", "Set milestone to daily, weekly or monthly")
-	flag.StringVar(&baseURL, "BaseURL", "dev.example.com", "Gitlab api base url")
-	flag.StringVar(&namespace, "Namespace", "someNamespace", "Namespace to use in Gitlab")
-	flag.StringVar(&project, "ProjectName", "someProject", "Project to use in Gitlab")
-	flag.IntVar(&advance, "Advance", 30, "Define timeframe to generate milestones in advance")
+	flag.StringVar(&token, "token", "jGWPwqQUuf37b", "Gitlab api key/token")
+	flag.StringVar(&timeInterval, "time-interval", "daily", "Set milestone to daily, weekly or monthly")
+	flag.StringVar(&baseURL, "base-url", "dev.example.com", "Gitlab api base url")
+	flag.StringVar(&namespace, "namespace", "someNamespace", "Namespace to use in Gitlab")
+	flag.StringVar(&project, "project", "someProject", "Project to use in Gitlab")
+	flag.IntVar(&advance, "advance", 30, "Define timeframe to generate milestones in advance")
 	flag.Parse() //Command Line Parsing Ends
 
 	// Initializing logger
 	LoggerSetup(os.Stdout)
 
-	milestoneData := createMilestoneData(advance, strings.ToLower(timeInterval))
-
 	// Validate baseURL scheme
 	URL, err := validateBaseURLScheme(baseURL)
 	if err != nil {
 		logger.Println(err)
 	}
 
-	// Calling getProjectID
-	baseURL = URL + "/api/v4"
-	projectID, err := getProjectID(baseURL, token, project, namespace)
+	// Check which API to use
+	api, err := checkAPI(URL, token, namespace, project)
 	if err != nil {
 		logger.Fatal(err)
-		// TODO: check for authentication error (currently it only says project not found)
-	}
-	err = createAndDisplayNewMilestones(baseURL, token, projectID, milestoneData)
-	if err != nil {
-		logger.Println(err)
 	}
-	editMilestones, err := getClosedMilestones(baseURL, token, projectID, milestoneData)
-	if err != nil {
-		logger.Println(err)
-	}
-	err = reactivateClosedMilestones(editMilestones, baseURL, token, projectID)
-	if err != nil {
-		logger.Println(err)
+	milestoneData := createMilestoneData(advance, strings.ToLower(timeInterval), api)
+
+	// Calling getProjectID
+	var newBaseURL, projectID string
+	switch {
+	case api == "gitlab":
+		newBaseURL = URL + "/api/v4"
+		projectID, err = getProjectID(newBaseURL, token, project, namespace, api)
+		if err != nil {
+			logger.Fatal(err)
+		}
+		err = createAndDisplayNewMilestones(newBaseURL, token, projectID, milestoneData, api)
+		if err != nil {
+			logger.Println(err)
+		}
+		editMilestones, err := getClosedMilestones(newBaseURL, token, projectID, milestoneData, api)
```

Comment:
Why editMilestones? Can't relate from the name to the content easily.

Reply:
Agreed - I renamed editMilestones to closedMilestones.